source/jormungandr/jormungandr/scenarios/experimental.py

```diff
@@ -575,192 +42,124 @@ def __init__(self):
-            elif len(g.origins_fallback[dep_mode]) == 0:
-                resp.append(self._make_error_response("no origin point", response_pb2.Error.no_origin))
-            elif len(g.destinations_fallback[arr_mode]) == 0:
-                resp.append(self._make_error_response("no destination point", response_pb2.Error.no_destination))
-            return resp
-        for r in resp:
+        logger.debug('request datetime: %s', request['datetime'])
+
+        requested_dep_modes = {mode for mode, _ in krakens_call}
+        requested_arr_modes = {mode for _, mode in krakens_call}
+        res = []
+
+        requested_orig = PlaceByUri(instance=instance, uri=request['origin'])
+        requested_dest = PlaceByUri(instance=instance, uri=request['destination'])
+
+        requested_orig_obj = get_entry_point_or_raise(requested_orig, request['origin'])
+        requested_dest_obj = get_entry_point_or_raise(requested_dest, request['destination'])
+
+        streetnetwork_path_pool = StreetNetworkPathPool(instance=instance)
+        period_extremity = PeriodExtremity(request['datetime'], request['clockwise'])
+
+        # we launch direct path asynchrnously
+        for mode in requested_dep_modes:
+            streetnetwork_path_pool.add_async_request(requested_orig_obj=requested_orig_obj,
+                                                      requested_dest_obj=requested_dest_obj,
+                                                      mode=mode,
+                                                      period_extremity=period_extremity,
+                                                      request=request,
+                                                      streetnetwork_path_type=StreetNetworkPathType.DIRECT)
+
+        # if max_duration(time to pass in pt) is zero, there is no need to continue,
+        # we return all direct path without pt
+        if request['max_duration'] == 0:
+            return [streetnetwork_path_pool.wait_and_get(requested_orig_obj=requested_orig_obj,
+                                                         requested_dest_obj=requested_dest_obj,
+                                                         mode=mode,
+                                                         period_extremity=period_extremity,
+                                                         streetnetwork_path_type=StreetNetworkPathType.DIRECT)
+                    for mode in requested_dep_modes]
+
+        orig_proximities_by_crowfly = ProximitiesByCrowflyPool(instance=instance,
+                                                               requested_place_obj=requested_orig_obj,
+                                                               modes=requested_dep_modes,
+                                                               request=request,
+                                                               streetnetwork_path_pool=streetnetwork_path_pool)
```

Comment:
Yep, the `streetnetwork_path_pool` is used to limit the max_duration for isochrone. 

Departure isochrone's max_duration depend on `direct_path` but arrival isochrone's  don't.   

IMO,  `streetnetwork_path_pool` is actually used and shared everywhere, renaming it will be more confusing